napari/components/layerlist.py

```diff
@@ -257,21 +257,39 @@ def _step_size_from_scales(self, scales):
     def _get_step_size(self, layer_extent_list):
         if len(self) == 0:
             return np.ones(self.ndim)
-        else:
-            scales = [extent.step for extent in layer_extent_list]
-            min_scales = self._step_size_from_scales(scales)
-            return min_scales
 
-    @cached_property
-    def extent(self) -> Extent:
-        """Extent of layers in data and world coordinates."""
-        extent_list = [layer.extent for layer in self]
+        scales = [extent.step for extent in layer_extent_list]
+        return self._step_size_from_scales(scales)
+
+    def get_extent(self, layers: List[Layer]) -> Extent:
```

Comment:
```suggestion
    def get_extent(self, layers: Sequence[Layer]) -> Extent:
```

Optional: I might consider passing something like `viewer.layers[0:2]`, which is a `LayerList` but not a `List`. `Sequence` is a type that captures both, but I guess you could even consider `Iterable` given the current implementation.